### ISSUE TITLE
Element Binding/Dynamo not properly deleting elements after closing and Re-Opening Revit "Project File" and Re-running DYN File

### ISSUE THREAD

**Issue by RonFS** [DynamoDS/Dynamo#5224]
I have been trying to troubleshoot this issue, and have narrowed it down to one specific situation. Any help/insight into this would be greatly appreciated.

Let's say I have, for example, opened a new Revit Project File and I have a DYN file that is reading point data from excel and breaking this data up to apply to an AdaptivePoints Component. If I:

1) Run the Script
          -Element ID's are generated
2) Save the Excel File (this will make Dynamo re-compute the script when I hit Run) and run the                                        script
          -Element ID's remain the same
3) "Save As" Excel file with a new name and run the script
          -Element ID's remain the same
4)Use different excel file with point data (just a previous design version, so the point data is different) and run the script
          -Element ID's remain the same
5)Save the Revit Project File and Dyn File, close and re-open the files, and run the script
          -Element ID's have changed
          -New Adaptive Components have been generated on top of the old ones and the old ones are
           remain          
 
The problem occurs when this Revit Project File, a model containing only the facade of a building, is linked back into the central model. Now that these facade elements have new element ID's, all annotation (DIMS, keynotes, etc.,) are blown away. As you can imagine this becomes a problem during the documentation process. Is this a unique issue or something that is currently being worked? I found this conversation that discusses a similar issue, (https://github.com/DynamoDS/DynamoRevit/pull/301) but sounds as the issue has been fixed. Running Revit 2015 and Dynamo 0.8.2.2392.

**Comment by RonFS** [comment]
I tried the same procedure on a very simple script, and it resulted in the same thing. When saving, closing and re-opening the component was not replaced. It was instantiated on the old one and the element ID was changed.
![element id not binding](https://cloud.githubusercontent.com/assets/8183746/9645526/6573146a-5189-11e5-8314-2634b856674f.jpg)

**Comment by riteshchandawar** [comment]
This seems to be a valid issue, Which version of Dynamo you are using currently?

If you are using 0.8.2 then Can you please try this workflow on 0.8.1 if possible?

Meanwhile I will investigate it at my end as well.

Thanks,
Ritesh

**Comment by riteshchandawar** [comment]
Hello @RonFS 

I tried your scenario at my end using our own Adaptive Component Sample. (Help>Samples>Revit>Revit_Adaptive Component Placement)
I am able to reproduce this on 0.8.1 and 0.8.2 but **on 0.8.3 it is not reproducible**.

I tried below steps.

* Saved Revit file without AC.
* Saved DYN file without running.
* Opened saved Revit file and execute AC script to create few AC.
* Saved both DYN and Revit and exited from Dynamo and Revit.
* Reopened Revit file with AC.
* Reopened Dynamo file and changed slider to increase/decrease number of AC. 
* It has updated corrected.

I tried above scenario multiple times with differnet number of AC and it was updating correctly.

Is it possible for you to try this out on daily build? (http://dynamobuilds.com/)

Let me know your findings.

Thanks,
Ritesh

**Comment by RonFS** [comment]
Hello @riteshchandawar,

Thank you for helping me to resolve this issue.

When running my test script described above I was using 0.8.2. I then updated to 0.8.3.503 and obtained the same results. I also took your recommendation to update to the latest daily build and also obtained the same results.

It took me a while to run the example file in the procedure you described, as every time I installed the latest daily build it would delete the sample files. I ended up copying the sample files on to my desktop and copying them back into my program files after installing the latest build.

After following your procedure with the sample files I still ended up with duplicate AdaptiveComps.

You seemed to have a methodical process for creating the association between RVT and DYN files. Is this necessary for maintaining the history of elements or you are simply outlining a potential situation? Does dynamo not maintain this history within the DYN file?

Also, as a side note, when running my test script I was not able to see the Points I was generating in Dynamo, only in Revit. I checked my view settings in Dynamo and they were correct. Do I have a bigger issue here?

Please let me know your thoughts.

Many Thanks,

Ron

**Comment by RonFS** [comment]
@riteshchandawar 

I am also willing to hold a quick goto meeting so we can be sure there is no miscommunication on the issue. It surprises me that you were able to not reproduce this issue in 0.8.3, yet I was.

**Comment by riteshchandawar** [comment]
Sure, let me know when you want to donit?

**Comment by RonFS** [comment]
Anytime before noon works for me. Just let me know and I will send you a zoom meeting link.

**Comment by riteshchandawar** [comment]
For me afternoons are prefarable. Can we do it tomorrow afternoon 3.00 Pm my time. (Singapore Time.)

Thanks,
Ritesh

**Comment by RonFS** [comment]
Ritesh,

I didn't think about the fact that you could actually be anywhere when I asked what time is convenient for you. I am in MST. 3pm Singapore time would be 1am my time. Can we try for 7-8am my time, or, 7-8am your time? I'm open to other suggestions, however, your afternoon is my early morning.

Thanks,

Ron

**Comment by RonFS** [comment]
Hi @riteshchandawar ,

Sorry I have been busy lately. Has anything else come of our previous conversations? Would you still be willing to set up a quick conference call?

Thanks,

Ron

**Comment by riteshchandawar** [comment]
You can set up somthing for this Thursday or Friday 7-8 Am my time.

Thanks,
Ritesh

**Comment by RonFS** [comment]
Sounds good. I will send a meeting invite for 7:30 AM your time.

Thanks,

Ron

**Comment by kronz** [comment]
@RonFS can you check this?  We did a lot of work on Adaptive Components in 0.9.1, this should be fixed

**Comment by RonFS** [comment]
Hi @kronz ,

We eventually found out this was an isolated issue. After trying on a different machine, the binding worked fine. I think this may have something to do with the fact that I was doing "clean" installs whenever I updated my dynamo. I think that in that process things got a little screwy (maybe certain things still existed in my registry which I did not look into).

Since then, I simply download the latest releases and install them without removing previous dynamo information from my system, and have not had a problem since.

However, the cause of that glitch was never really determined.

Cheers,

Ron
